src/lib/storage/segment_encoding_utils.cpp

```diff
@@ -74,4 +75,24 @@ VectorCompressionType parent_vector_compression_type(const CompressedVectorType
   Fail("Invalid enum value.");
 }
 
+SegmentEncodingSpec auto_select_segment_encoding_spec(const DataType& type, const bool is_unique) {
+  switch (type) {
+    case DataType::Int:
+    case DataType::Long:
+      return SegmentEncodingSpec{EncodingType::FrameOfReference};
+    case DataType::String:
+    case DataType::Double:
+    case DataType::Float:
+      if (is_unique) {
+        return SegmentEncodingSpec{EncodingType::Unencoded};
+      } else {
+        return SegmentEncodingSpec{EncodingType::Dictionary};
+      }
+    case DataType::Null:
+      return SegmentEncodingSpec{EncodingType::RunLength};
+    default:
+      Fail("Unknown DataType when trying to select encoding for column");
+  }
+}
+
```

Comment:
@Bouncner Feel free to suggest other encoding selections.

@tjeyy I had a problem where gcc would not compile without the default case. Clang did compile. Is this maybe part of #2692?

Reply:
It's a known issue of GCC and independent of our warning selection / #2692  :/